enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,335 @@
+to obtain basic product version details for a cluster.
+
+### User Stories
+
+As a console user, I want to see the version of OCP I am using so that
+I can refer to the correct version of the documentation and API
+references.
+
+As a console developer, I want to be able to obtain the version of OCP
+that I am using so that the console app can use different
+implementation choices for behaviors that cannot be discovered by
+probing the Kubernetes API versions supported.
+
+As an authenticated API caller, I want to obtain the product name and
+version of the OpenShift variant I am communicating with so that I can
+use it to make decisions on the client side.
+
+As an application developer, I want to be able to determine the
+Kubernetes distribution the application is being installed into at
+runtime so that I only need to implement one set of installation
+logic.
+
+As an OpenShift user without admin privileges, I want to find
+information about the version of the product I am using in a
+consistent way, regardless of whether I am using standalone OpenShift,
+HyperShift, or MicroShift.
+
+### Goals
+
+* Use a standard Kubernetes type, to keep access easy for arbitrary
+  API clients. This also lets us avoid introducing a new CRD, so we do
+  not increase the overhead for MicroShift.
+* Define a standard location for version information that all
+  OpenShift variants can support.
+
+### Non-Goals
+
+* This change does not introduce a new way for a cluster admin to
+  update the version of a cluster (i.e., to perform an upgrade).
+* This change does not introduce a new way for a client to discover
+  that a cluster is being upgraded.
+* This change does not introduce a new way for a client to discover
+  the cluster topology (single-node, HA, external control-plane, etc.).
+* This change does not introduce a new way for a client to discover
+  other properties of the cluster that might currently only be visible
+  using OpenShift-specific APIs.
+* This change does not introduce a way for a cluster consumer to tell
+  the difference between ROSA, ARO, OSD, HyperShift, or standalone
+  OpenShift clusters because we want those to appear the same from the
+  workload's perspective.
+
+## Proposal
+
+Because we want arbitrary clients to be able to read and understand
+the data without requiring adopting an OpenShift-specific API, we will
+use a standard Kubernetes ConfigMap.
+
+Because we want all OpenShift variants to publish the information in
+the same way and that the data should not be writable by API callers,
+we will specify the name and namespace for the ConfigMap.
+
+Because MicroShift exposes a limited number of namespaces by default,
+we need to choose from one of those. The most suitable available
+namespace is `openshift`, because it is available in all OpenShift
+clusters, regardless of topology or form-factor and is readable by
+authenticated users by design and includes other cluster-wide details.
```

Comment:
Well, it looks like `openshift` is _not_ available on MicroShift. Should we create it, or use `kube-public`?

```
$ oc get namespace
NAME                           STATUS   AGE
default                        Active   32m
kube-node-lease                Active   32m
kube-public                    Active   32m
kube-system                    Active   32m
openshift-controller-manager   Active   32m
openshift-dns                  Active   32m
openshift-infra                Active   32m
openshift-ingress              Active   32m
openshift-ovn-kubernetes       Active   32m
openshift-service-ca           Active   32m
openshift-storage              Active   32m
```